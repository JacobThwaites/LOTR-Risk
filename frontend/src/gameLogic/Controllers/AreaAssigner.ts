import { Player } from "../Models/Player";
import { AreaType } from "../Models/AreaType";

export class AreaAssigner {
  private players: Array<Player>;
  constructor(players: Array<Player>) {
    this.players = players;
  }

  assignAreas(areaLists: Array<AreaType[]>) {
    for (let i = 0; i < this.players.length; i++) {
      for (let j = 0; j < areaLists[i].length; j++) {
        this.addAreaToPlayer(areaLists[i][j], this.players[i]);
      }
      
    }
  }

  private addAreaToPlayer(area: AreaType, player: Player): void {
    player.addArea(area);
    area.setPlayer(player);
  }
}