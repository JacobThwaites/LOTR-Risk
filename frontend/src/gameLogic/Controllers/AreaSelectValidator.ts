import { Adjacencies } from "../Enums/AreaAdjacencies";
import { AreaType } from "../Models/AreaType";
import { Player } from "../Models/Player";

export function getAreasForUnitManeuvre(area: AreaType, player: Player): Array<AreaType> {
    const connected: Array<AreaType> = [];
    const visited: any = {};


    function dfs(area: AreaType) {
        const areaName = area.getName();
        if (visited[areaName]) {
            return;
        }

        visited[areaName] = true;

        if (!player.ownsArea(area)) {
            return;
        }

        connected.push(area);
        const adjacencies = Adjacencies[areaName];

        for (let i = 0; i < adjacencies.length; i++) {
            dfs(adjacencies[i]);
        }
    }

    dfs(area);
    
    return connected;
}