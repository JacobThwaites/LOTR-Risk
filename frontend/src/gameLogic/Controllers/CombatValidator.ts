import { Combat } from '../Enums/Combat'; 
import { AreaType } from '../Models/AreaType';
const { MAX_ATTACKING_DICE } = Combat;

export class CombatValidator {
    public static isCombatValid(attackingArea: AreaType, attackingDice: number): boolean {
        return this.validateAttackingDice(attackingArea, attackingDice);
    }

    private static validateAttackingDice(attackingArea: AreaType, attackingDice: number): boolean {
        const attackingAreaUnits = attackingArea.getUnits();
        return (
            attackingAreaUnits > attackingDice &&
            attackingDice <= MAX_ATTACKING_DICE &&
            attackingDice > 0
        );
    }
}