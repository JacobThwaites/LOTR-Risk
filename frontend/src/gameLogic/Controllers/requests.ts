export async function saveGame(numPlayers: number, playerAreas: Array<string>, playerName: string) {
    const players = formatPlayerData(playerAreas, playerName);

    try {
        const body = {
            numPlayers: numPlayers,
            players: players
        }

        return fetch('http://localhost:8000/api/game', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(body)
        });
    } catch (err) {
        console.error(err);
        return;
    }
}

function formatPlayerData(playerAreas: Array<string>, playerName: string): Array<{ name: string, areas: string }> {
    const players = [];

    const firstPlayer = {
        name: playerName,
        areas: playerAreas[0]
    }
    players.push(firstPlayer);

    for (let i = 1; i < playerAreas.length; i++) {
        const player = { name: '', areas: playerAreas[i] };
        players.push(player)
    }

    return players;
}

export async function getGame(gameID: string) {
    try {
        return fetch(`http://localhost:8000/api/game/${gameID}`, {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
        });
    } catch (err) {
        console.error(err);
        return;
    }
}

export async function addUserIdToPlayer(playerID: string, userID: string) {
    try {
        const res = await fetch(`http://localhost:8000/api/player/${playerID}`, {
            method: 'PATCH',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ userID })
        });

        if (!res.ok) {
            throw new Error("Faiedl to update player userID");
        }

        return true;

    } catch (err: any) {
        console.error(err.message);
        return false;
    }
}