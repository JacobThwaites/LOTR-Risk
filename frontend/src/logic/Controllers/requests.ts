export async function saveGame(numPlayers: number, playerAreas: Array<string | null>) {
  try {
    const body = {
        numPlayers: numPlayers,
        player1Areas: playerAreas[0],
        player2Areas: playerAreas[1],
        player3Areas: playerAreas[2],
        player4Areas: playerAreas[3]
    }

      return fetch('http://localhost:8000/api/game', {
          method: 'POST',
          headers: {
              'Accept': 'application/json',
              'Content-Type': 'application/json'
          },
          body: JSON.stringify(body)
      });
  } catch(err) {
      console.error(err);
      return;
  }
}

export async function getGame(gameUuid: string) {
    try {
        return fetch(`http://localhost:8000/api/game/${gameUuid}`, {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
        });
    } catch(err) {
        console.error(err);
        return;
    }
  }