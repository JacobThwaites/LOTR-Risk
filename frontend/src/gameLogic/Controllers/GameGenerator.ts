import { Player } from '../Models/Player';
import { Game } from '../Models/Game'; 
import { AreaAssigner } from './AreaAssigner';
import { PlayerGenerator } from './PlayerGenerator';
import { AreaType } from '../Models/AreaType';

export class GameGenerator {
    public static generateGame(areaLists: Array<AreaType[]>, playerIDs: Array<number>): Game {
        const MAX_TURNS = 30;
        const players = this.generatePlayers(playerIDs);
        const areaAssigner = new AreaAssigner(players);
        areaAssigner.assignAreas(areaLists);
        const game = new Game(players, MAX_TURNS)
        game.assignStartingUnits();
        
        return game;
    }

    private static generatePlayers(playerIDs: Array<number>): Player[] {
        const playerGenerator = new PlayerGenerator(playerIDs);
        const players = playerGenerator.generatePlayers();
        return players;
    } 
}