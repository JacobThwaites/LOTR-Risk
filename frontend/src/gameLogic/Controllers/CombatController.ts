import { AreaType } from '../Models/AreaType';
import { CombatValidator } from './CombatValidator';

export class CombatController {
    private attackingArea: AreaType;
    private defendingArea: AreaType;
    constructor(attackingArea: AreaType, defendingArea: AreaType) {
        this.attackingArea = attackingArea;
        this.defendingArea = defendingArea;
    }

    getCombatResults(numAttackingDice: number, numDefendingDice: number) {
        const results = [];
        const attackingDice = this.rollDice(numAttackingDice);
        const defendingDice = this.rollDice(numDefendingDice);
        
        const firstResult = this.firstDiceCombat(attackingDice[0], defendingDice[0]);
        results.push(firstResult);

        if (numDefendingDice > 1) {
            const secondResult = this.secondDiceCombat(attackingDice[0], defendingDice[0]);
            results.push(secondResult);
        }

        return results;
    }

    firstDiceCombat(attackingDice: number, defendingDice: number) {
        const defenderBonus = this.getDefenderDiceBonus(this.defendingArea);
        const attackerBonus = this.getAttackerDiceBonus(this.attackingArea);
        // TODO: remove + 6
        const attackerScore = attackingDice + attackerBonus + 6;
        const defenderScore = defendingDice + defenderBonus - 6;
        if (attackerScore > defenderScore) {
            return 'attacker';
        } else {
            return 'defender';
        }
    }

    secondDiceCombat(attackerScore: number, defenderScore: number) {
        if (attackerScore > defenderScore) {
            return 'attacker';
        } else {
            return 'defender';
        }
    }

    handleResults(results: string[]) {
        for (let i = 0; i < results.length; i++) {
            if (results[i] === 'attacker') {
                this.removeUnitsFromLoser(this.defendingArea);
            } else {
                this.removeUnitsFromLoser(this.attackingArea);
            }  
        }

        this.checkDefendingUnitsRemaining();
    }

    removeUnitsFromLoser(losingArea: AreaType) {
        const losingPlayer = losingArea.getPlayer();
        losingPlayer!.removeUnits(1);
        losingArea.removeUnits(1);
    }

    isCombatValid(attackingDice: number, defendingDice: number): boolean {
        const combatValidator = new CombatValidator(this.attackingArea, this.defendingArea);
        return combatValidator.isCombatValid(attackingDice, defendingDice);
    } 

    rollDice(numberOfDice: number): Array<number> {
        let numbersRolled = [];
        for (let i = 0; i < numberOfDice; i++) {
            const number = this.getDiceRoll();
            numbersRolled.push(number); 
        }
        
        return numbersRolled.sort().reverse();
    }

    getDiceRoll(): number {
        const numberRolled = Math.floor(Math.random() * 6) + 1; 
        return numberRolled;
    }

    getDefenderDiceBonus(defendingArea: AreaType): number {
        let bonus = 0;
        bonus += defendingArea.getDefendingBonus();

        if (defendingArea.getHasLeader()) {
            bonus += 1;
        }
        return bonus;
    }

    getAttackerDiceBonus(attackingArea: AreaType): number {
        let bonus = 0;
        if (attackingArea.getHasLeader()) {
            bonus += 1;
        }
        return bonus;
    }

    checkDefendingUnitsRemaining() {
        const attacker = this.attackingArea.getPlayer();

        if (!attacker) {
            return;
        }

        if (this.defendingArea.getUnits() < 1) {
            attacker.addArea(this.defendingArea);
            this.defendingArea.setPlayer(attacker);
        }
    }
}