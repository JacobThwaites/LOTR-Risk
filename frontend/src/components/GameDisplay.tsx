import React, { useEffect, useState } from "react";
import { GameGenerator } from "../gameLogic/Controllers/GameGenerator";
import { CombatController } from "../gameLogic/Controllers/CombatController";
import { ReinforcementController } from "../gameLogic/Controllers/ReinforcementController";
import { UnitManeuverController } from "../gameLogic/Controllers/UnitManeuverController";
import CombatHandler from "./CombatHandler";
import UnitManeuverHandler from "./UnitManeuverHandler";
import Map from "./Map";
import EndTurnButton from "./buttons/EndTurnButton";
import ReinforcementsModal from "./ReinforcementsModal";
import GameOverModal from "./GameOverModal";
import TurnInformation from "./TurnInformation";
import { Combat } from '../gameLogic/Enums/Combat';
import { CombatValidator } from "../gameLogic/Controllers/CombatValidator";
import { Game } from "../gameLogic/Models/Game";
import { Area } from "../gameLogic/Models/Area";
import { useParams } from "react-router";
import { getGame } from "../gameLogic/Controllers/requests";
import { getAreas } from "../utils/playerAreaParser";
import WebSocketHandler, { GameEventType } from "../utils/WebSocketHandler";
import { Areas } from "../gameLogic/Enums/Areas";
import { AreaType } from "../gameLogic/Models/AreaType";

type PlayerResponseType = {
    "id": string,
    "name": string,
    "areas": string,
    "gameID": string
}

function GameDisplay(): JSX.Element {
    const [game, setGame] = useState<Game | null>(null);
    const [attackingArea, setAttackingArea] = useState<Area | null>(null);
    const [defendingArea, setDefendingArea] = useState<Area | null>(null);
    const [attackingDice, setAttackingDice] = useState<number>(1);
    const [defendingDice, setDefendingDice] = useState<number>(1);
    const [shouldDisplayUnitManeuverButton, setShouldDisplayUnitManeuverButton] = useState<boolean>(false);
    const [shouldDisplayReinforcementsModal, setShouldDisplayReinforcementsModal] = useState<boolean>(false);
    const [shouldHandleStartingReinforcements, setShouldHandleStartingReinforcements] = useState<boolean>(true);
    const [areaToMoveUnits, setAreaToMoveUnits] = useState<Area | null>(null);
    const [areaToReceiveUnits, setAreaToReceiveUnits] = useState<Area | null>(null);
    const [unitsToMove, setUnitsToMove] = useState<number>(0);
    const [isGameOver, setIsGameOver] = useState<boolean>(false);
    const [rerender, setRerender] = useState(false);
    const { gameID } = useParams<{ gameID: string }>();
    const [socket, setSocket] = useState<WebSocket | null>(null);
    const [socketHandler, setSocketHandler] = useState<WebSocketHandler | null>(null);

    useEffect(() => {
        setupGame();
    }, [])

    useEffect(() => {
        setSocket(new WebSocket(`ws://localhost:8001/api/game/${gameID}`));
        setSocketHandler(new WebSocketHandler(gameID));
    
        return () => {
          if (socket) {
            socket.close();
          }
        };
      }, [gameID]);

      useEffect(() => {
        if (!socket) {
          return;
        }
    
        socket.onopen = () => {
          console.log('Connected to the WebSocket server');
        };
    
        socket.onmessage = (event) => {
            const messageData = JSON.parse(event.data);

            if (socketHandler!.isMessageAlreadyProcessed(messageData.id)) {
                return;
            }

            socketHandler!.previousMessageUUID = messageData.id;

            if (messageData.type === GameEventType.COMBAT) {
                const attackingArea = Areas[messageData.attackingArea];
                const defendingArea = Areas[messageData.defendingArea];
                setAttackingArea(attackingArea);
                setDefendingArea(defendingArea);
            } else if (messageData.type === GameEventType.CLEAR_SELECTED_AREAS) {
                clearSelectedAreas();
            } else if (messageData.type === GameEventType.STARTING_REINFORCEMENT) {
                const area = Areas[messageData.areaName];
                handleStartingReinforcements(area);
            } else if (messageData.type === GameEventType.COMBAT_RESULTS) {
                updateAreasAfterCombat(messageData.attackingArea, messageData.defendingArea, messageData.results);
            }
        };
      }, [socket, game]);

    async function setupGame() {
        const res = await getGame(gameID);
        const json = await res!.json(); 
        const areaNames = formatPlayerAreas(json.data.players);
        const areas = getAreas(areaNames);
        const gameGenerator = createGameGenerator(json.data.num_players);
        const game = gameGenerator.generateGame(areas);
        setGame(game);
        setShouldDisplayReinforcementsModal(true);
    }

    function formatPlayerAreas(playerData: Array<PlayerResponseType>): Array<string> {
        const playerAreas = [];

        for (let i = 0; i < playerData.length; i++) {
            playerAreas.push(playerData[i].areas);   
        }

        return playerAreas;
    }

    function createGameGenerator(numberOfPlayers: number): GameGenerator {
        const maxTurns = 30;
        const gameGenerator = new GameGenerator(numberOfPlayers, maxTurns);

        return gameGenerator;
    }

    function onAreaSelect(area: Area): void {
        if (shouldHandleStartingReinforcements) {
            socketHandler!.sendStartingReinforcement(area.getName());
        } else if (shouldDisplayReinforcementsModal) {
            addReinforcements(area);
        } else {
            setAreaForCombat(area);
        }
    }

    function handleStartingReinforcements(area: Area): void {
        const currentPlayer = game!.getCurrentPlayer();

        if (game!.playersHaveReinforcements()) {
            addReinforcements(area);

            if (currentPlayer!.getReinforcements() < 1) {
                game!.changeCurrentPlayer();
                setGame(game);
                setRerender(!rerender);
            }

            return;
        }

        game!.changeCurrentPlayer();
        setShouldDisplayReinforcementsModal(false);
        setShouldHandleStartingReinforcements(false)
    }


    function addReinforcements(area: Area): void {
        const reinforcementController = createReinforcementController();
        reinforcementController.addReinforcements(area);
        setRerender(!rerender);

        if (shouldHideReinforcementsModal()) {
            setShouldDisplayReinforcementsModal(false);
        }
    }

    function createReinforcementController(): ReinforcementController {
        const currentPlayer = game!.getCurrentPlayer();
        const reinforcementController = new ReinforcementController(currentPlayer!);
        return reinforcementController;
    }

    function shouldHideReinforcementsModal(): boolean {
        const currentPlayer = game!.getCurrentPlayer();
        const reinforcementsAvailable = currentPlayer!.getReinforcements();
        return reinforcementsAvailable < 1 && !shouldHandleStartingReinforcements
    }

    function setAreaForCombat(area: Area): void {
        if (attackingArea === area) {
            socketHandler!.sendClearAreaSelection();
        } else if (defendingArea === area) {
            setDefendingArea(null);
        } else if (attackingArea !== null) {
            setDefendingArea(area);
            socketHandler!.sendCombatInfo(attackingArea.getName(), area.getName());
        } else {
            setAttackingArea(area);
        }
    }

    function clearSelectedAreas() {
        setAttackingArea(null);
        setDefendingArea(null); 
    }

    async function onCombatButtonClick() {
        handleCombat(attackingArea!, defendingArea!);
    }

    function handleCombat(attackingArea: AreaType, defendingArea: AreaType): void {
        const combatController = new CombatController(
            attackingArea!,
            defendingArea!
        );

        const results = combatController!.getCombatResults(attackingDice, defendingDice);
        socketHandler!.sendCombatResults(attackingArea.getName(), defendingArea.getName(), results);
    }

    async function updateAreasAfterCombat(attackingAreaName: string, defendingAreaName: string, results: string[]) {
        const attackingArea = Areas[attackingAreaName];
        const defendingArea = Areas[defendingAreaName];
        const combatController = new CombatController(
            attackingArea,
            defendingArea
        );
        
        combatController.handleResults(results);

        if (!defendingArea.hasUnitsRemaining()) {
            await setStateForManeuvers(attackingArea, defendingArea);
        }

        resetCombatState();
    }

    function setStateForManeuvers(attackingArea: any, defendingArea: any): void {
        setShouldDisplayUnitManeuverButton(true);
        setAreaToMoveUnits(attackingArea);
        setAreaToReceiveUnits(defendingArea);
    }

    function resetCombatState(): void {
        setAttackingArea(null);
        setDefendingArea(null);
        setAttackingDice(1);
        setDefendingDice(1);
    }

    function onEndTurnClick(): void {
        game!.handleNewTurn();
        setShouldDisplayReinforcementsModal(true);
        resetCombatState();
        checkIfGameOver();
    }

    function checkIfGameOver(): void {
        const maxTurnsReached = game!.checkMaxTurnsReached();

        if (maxTurnsReached) {
            setIsGameOver(true);
        }
    }

    function onMoveUnits(): void {
        const unitManeuverController = createUnitManeuverController();
        const areUnitsMoved = unitManeuverController.handleManeuver(unitsToMove);

        if (areUnitsMoved) {
            resetManeuverState();
        }
    }

    function createUnitManeuverController(): UnitManeuverController {
        const unitManeuverController = new UnitManeuverController(
            areaToMoveUnits!,
            areaToReceiveUnits!
        );

        return unitManeuverController;
    }

    function resetManeuverState(): void {
        setShouldDisplayUnitManeuverButton(false);
        setAreaToMoveUnits(null);
        setAreaToReceiveUnits(null);
        setUnitsToMove(0);
    }

    function getMaxAttackingDice(): number {
        const { MAX_ATTACKING_DICE } = Combat;
        return Math.min(MAX_ATTACKING_DICE, attackingArea!.getUnits() - 1);
    }

    function getMaxDefendingDice(): number {
        const { MAX_DEFENDING_DICE } = Combat;
        return Math.min(attackingDice, defendingArea!.getUnits(), MAX_DEFENDING_DICE);
    }

    function isCombatButtonClickable(): boolean {
        const combatValidator = new CombatValidator(attackingArea!, defendingArea!);
        const isValid = combatValidator.isCombatValid(attackingDice, defendingDice);
        return isValid;
    }

    function isMoveUnitsButtonDisabled(): boolean {
        return unitsToMove < 1;
    }

    function isEndTurnButtonDisabled(): boolean {
        return (
            shouldDisplayUnitManeuverButton ||
            shouldDisplayReinforcementsModal ||
            shouldHandleStartingReinforcements
        );
    }

    if (!game) {
        return (<></>);
    }

    const currentPlayer = game!.getCurrentPlayer();

    return (
        <div id='game-display'>
            <Map
                attackingArea={attackingArea}
                defendingArea={defendingArea}
                attackingDice={attackingDice}
                currentPlayer={currentPlayer!}
                onAreaSelect={onAreaSelect}
            />
            <TurnInformation
                turnsRemaining={game!.getTurnsRemaining()}
                playerName={currentPlayer!.getName()}
            />
            {attackingArea && defendingArea && (
                <CombatHandler
                    attackingDice={attackingDice}
                    defendingDice={defendingDice}
                    maxAttackingDice={getMaxAttackingDice()}
                    maxDefendingDice={getMaxDefendingDice()}
                    onCombatButtonClick={onCombatButtonClick}
                    setAttackingDice={setAttackingDice}
                    setDefendingDice={setDefendingDice}
                    isCombatButtonClickable={isCombatButtonClickable()}
                />
            )}
            {shouldDisplayUnitManeuverButton && (
                <UnitManeuverHandler
                    max={areaToMoveUnits!.getUnits() - 1}
                    unitsToMove={unitsToMove}
                    onMoveUnits={onMoveUnits}
                    setUnitsToMove={setUnitsToMove}
                    isButtonDisabled={isMoveUnitsButtonDisabled()}
                />
            )}
            {shouldDisplayReinforcementsModal && (
                <ReinforcementsModal
                    reinforcementsAvailable={currentPlayer!.getReinforcements()}
                />
            )}
            <EndTurnButton
                onEndTurnClick={onEndTurnClick}
                isDisabled={isEndTurnButtonDisabled()}
            />
            {isGameOver && (
                <GameOverModal />
            )}
        </div>
    );
}

export default GameDisplay;