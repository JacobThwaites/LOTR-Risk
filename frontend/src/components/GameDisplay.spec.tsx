import '@testing-library/jest-dom'
import React from 'react';
import { render, screen, fireEvent, act, waitFor } from '@testing-library/react' 
import GameDisplay from './GameDisplay';
import { useParams, useLocation } from 'react-router';
import { addUserIDToGame } from '../gameLogic/Controllers/requests';

jest.mock('../gameLogic/Controllers/requests');
jest.mock('react-router');

it('renders Waiting for players page after game load', async () => {
    const mockGameID = '1234';
    (useParams as jest.Mock).mockReturnValue({gameID: mockGameID});
    (useLocation as jest.Mock).mockReturnValue({location: {state: {gameType: 'local'}}});
    const mockPlayer = {colour: 'Black', reinforcements: 2, areas:[]};
    const mockResponse = {
        ok: true, json: async () => ({ data: { uuid: mockGameID, players: [mockPlayer], maxTurns: 1 } }),
    };
    (addUserIDToGame as jest.Mock).mockResolvedValueOnce(mockResponse);
    
    let container: any;
    await act(() => {
        container = render(<GameDisplay />).container;
    })
    
    expect(addUserIDToGame).toHaveBeenCalledWith(mockGameID);
    const waitingForPlayersScreen = await screen.findByTestId('waiting-for-players-screen');
    expect(waitingForPlayersScreen).toBeTruthy();
    expect(container).toMatchSnapshot();
});
