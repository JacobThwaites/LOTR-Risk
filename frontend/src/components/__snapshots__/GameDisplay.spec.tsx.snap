// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Waiting for players page after game load 1`] = `
<div>
  <div
    data-testid="waiting-for-players-screen"
    id="waiting"
  >
    <svg
      id="waiting-image"
    >
      ring-loading.svg
    </svg>
    <h1>
      Waiting for 
      1
       more players to join...
    </h1>
    <h1>
      Share this url to allow others to join: 
      http://localhost/
    </h1>
  </div>
</div>
`;
